pkg/unit1_test.go

```diff
@@ -26,3 +26,48 @@ func TestToAndFromProtoMsg(t *testing.T) {
 	assert.Equal(t, fObj_proto.GetFA(), "")
 
 }
+
+func TestRandomPatternForAllFormat(t *testing.T) {
+	config := openapiart.NewPrefixConfig()
+	config.SetA("asdf").SetB(12.2).SetC(1).SetH(true).SetI([]byte{1, 0, 0, 1, 0, 0, 1, 1})
+	config.RequiredObject().SetEA(1).SetEB(2)
+	config.SetIeee8021Qbb(true)
+	config.SetFullDuplex100Mb(2)
+	config.SetResponse(openapiart.PrefixConfigResponse.STATUS_200)
+	config.E().SetEA(1.1).SetEB(1.2).SetMParam1("Mparam1").SetMParam2("Mparam2")
+	config.IntegerPattern().Integer().Random()
+	config.Ipv4Pattern().Ipv4().Random()
+	config.Ipv6Pattern().Ipv6().Random()
+	config.MacPattern().Mac().Random()
+
+	json, err := config.Marshal().ToJson()
+	assert.Nil(t, err)
+
+	config2 := openapiart.NewPrefixConfig()
+	err = config2.Unmarshal().FromJson(json)
+	assert.Nil(t, err)
+
+	pat1 := config2.IntegerPattern().Integer().Random()
+	assert.Equal(t, pat1.Min(), uint32(0))
+	assert.Equal(t, pat1.Max(), uint32(255))
+	assert.Equal(t, pat1.Count(), uint32(1))
+	assert.Equal(t, pat1.Seed(), uint32(1))
+
+	pat2 := config2.Ipv4Pattern().Ipv4().Random()
+	assert.Equal(t, pat2.Min(), "0.0.0.0")
+	assert.Equal(t, pat2.Max(), "255.255.255.255")
+	assert.Equal(t, pat2.Count(), uint32(1))
+	assert.Equal(t, pat2.Seed(), uint32(1))
+
+	pat3 := config2.Ipv6Pattern().Ipv6().Random()
+	assert.Equal(t, pat3.Min(), "::")
+	assert.Equal(t, pat3.Max(), "::")
```

Comment:
is this expected `Max` value for IPv6 (e,g, `::`)

Reply:
corrected it